gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1390,6 +1435,87 @@ protected void configureRestModelGeneration(Project project, SourceSet sourceSet
     });
   }
 
+  protected void configurePegasusSchemaSnapshotGeneration(Project project, SourceSet sourceSet)
```

Comment:
I feel this task function and the other function below `configurePegasusExtensionSchemaSnapshotGeneration`
share some common logic, would it be better to factor some common logic together to behave the DIY(Dont repeat yourself) pattern?

Reply:
I thought about it. Then I thought that generateExtensionSchemaSnapshot and generateSchemaSnapshot are two parallel code paths. Both of them call GeneratePegasusSnapshotTask to generate the snapshots. But the purposes of two tasks are kind of different. 
Having separate generatePegasusSchemaSnapshot task and generatePegasusExtensionSchemaSnapshot task makes future customized configuration easier.

Reply:
Abstracted common logic to helper methods.